src/components/pagination/pagination.jsx

```diff
@@ -0,0 +1,174 @@
+import { createContext, useContext, forwardRef } from 'react';
+import { cn } from '@/utilities/functions';
+import { ChevronLeft, ChevronRight } from 'lucide-react';
+import { disabledClassNames, sizeClassNames } from './component-style';
+import Button from '../button';
+
+const PageContext = createContext();
+const usePageContext = () => useContext( PageContext );
+
+const Pagination = ( {
+	size = 'sm',
+	disabled = false,
+	children,
+	className,
+	...props
+} ) => (
+	<PageContext.Provider value={ { size, disabled } }>
+		<nav
+			role="navigation"
+			aria-label="pagination"
+			className={ cn(
+				'flex w-full justify-center m-0 box-border',
+				className
+			) }
+			{ ...props }
+		>
+			{ children }
+		</nav>
+	</PageContext.Provider>
+);
+Pagination.displayName = 'Pagination';
+
+const PaginationContent = forwardRef( ( { className, ...props }, ref ) => {
+	return (
+		<ul
+			ref={ ref }
+			className={ cn(
+				'm-0 p-0 w-full flex justify-center flex-row items-center gap-1',
+				'list-none',
+				className
+			) }
+			{ ...props }
+		/>
+	);
+} );
+PaginationContent.displayName = 'Pagination.Content';
+
+const PaginationItem = forwardRef(
+	( { isActive = false, href = '', className, children, ...props }, ref ) => {
+		const { disabled } = usePageContext();
+		return (
+			<li
+				ref={ ref }
+				tabIndex={ 0 }
+				className={ cn( 'flex', disabled && disabledClassNames.general ) }
+			>
+				<PaginationButton
+					isActive={ isActive }
+					disabled={ disabled }
+					className={ className }
+					href={ href }
+					{ ...props }
+				>
+					{ children }
+				</PaginationButton>
+			</li>
+		);
+	}
+);
+PaginationItem.displayName = 'Pagination.Item';
+
+const PaginationButton = ( {
+	icon = null,
+	isActive = false,
+	href = '',
+	children,
+	className,
+	...props
+} ) => {
+	const { size, disabled } = usePageContext();
+
+	return (
+		<a
+			href={ href }
+			className="text-none no-underline bg-transparent p-0 m-0 border-none"
+		>
```

Comment:
If I want to set a `target` or `rel` how do I do that?

Reply:
Added ✅